pkg/service/service.go

```diff
@@ -262,11 +262,16 @@ func (s *Service) UpsertService(
 	}
 
 	// Only add a HealthCheckNodePort server if this is a service which may
-	// only contain local backends.
+	// only contain local backends (i.e. it has externalTrafficPolicy=Local)
 	if onlyLocalBackends {
 		localBackendCount := len(backendsCopy)
 		s.healthServer.UpsertService(lb.ID(svc.frontend.ID), svc.svcNamespace, svc.svcName,
 			localBackendCount, svc.svcHealthCheckNodePort)
+	} else if svc.svcHealthCheckNodePort == 0 {
+		// Remove the health check server in case this service used to have
+		// externalTrafficPolicy=Local with HealthCheckNodePort in the previous
+		// version, but not anymore.
+		s.healthServer.DeleteService(lb.ID(svc.frontend.ID))
```

Comment:
Is it OK not to check whether the health check node port was previously set?

Reply:
Yes, deleting is idempotent.